### PR TITLE
[MPS] Add tensor::index_put op

### PR DESCRIPTION
**Pull request by kulinseth**

### CODE REVIEW

**Review by malfet — CHANGES_REQUESTED** [review]
Lots of lint errors, and code does not compile

File: torch/testing/_internal/common_device_type.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
     skipCUDANonDefaultStreamIf, TEST_WITH_ASAN, TEST_WITH_UBSAN, TEST_WITH_TSAN, \
     IS_SANDCASTLE, IS_FBCODE, IS_REMOTE_GPU, IS_WINDOWS, \
     _TestParametrizer, compose_parametrize_fns, dtype_name, \
-    TEST_WITH_MIOPEN_SUGGEST_NHWC, NATIVE_DEVICES, skipIfTorchDynamo
+    TEST_WITH_MIOPEN_SUGGEST_NHWC, NATIVE_DEVICES, skipIfTorchDynamo, TEST_WITH_MPS
```
[the remark this answers is not in the document]
Comment (DenisVieriu97) [reply torch/testing/_internal/common_device_type.py]: Yes, I think that part of code got accidentally checked-in during rebase - I've removed it 👍

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**